File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.ts
function maxAscendingSum(nums: number[]): number {
    const n = nums.length
    if(n == 0) return 0;
    if(n == 1) return nums[0];

    let maxSum = nums[0]
    let sum = nums[0]

    for(let i = 1; i < n; i++) {
        if(nums[i - 1] >= nums[i]) sum = 0;
        sum += nums[i]
        maxSum = Math.max(maxSum, sum) 
    }
    return maxSum
};

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.go
func maxAscendingSum(nums []int) int {
	n := len(nums)

	if n == 0 {
		return 0
	}
	if n == 1 {
		return nums[0]
	}

	sum, maxSum := nums[0], nums[0]

	for i := 1; i < len(nums); i++ {
		if nums[i-1] >= nums[i] {
			sum = 0
		}
		sum += nums[i]
		maxSum = max(maxSum, sum)
	}

	return maxSum
}